src/bundle.rs

```diff
@@ -381,7 +388,18 @@ impl<T: Authorization, V: Copy + Into<i64>> Bundle<T, V> {
                 ValueSum::from_raw(self.value_balance.into()),
                 ValueCommitTrapdoor::zero(),
                 AssetId::native(),
-            ))
+            )
+            - self
+                .assets_burnt
+                .iter()
+                .map(|(asset, value)| {
+                    ValueCommitment::derive(
+                        ValueSum::from_raw(*value),
+                        ValueCommitTrapdoor::zero(),
+                        *asset,
+                    )
+                })
+                .sum::<ValueCommitment>())
```

Comment:
This is the same functionality like in `builder.rs:440`. It was acceptable previously because it was simple but now we should extract it and reuse.

Reply:
done